pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			garbagecollection.CollectDeadMemberServices(c.cluster.Name, c.cluster.Namespace, c.config.KubeCli, c.members, c.logger)
```

Comment:
add a todo? i do not feel good about letting GC tightly coupled with reconciliation loop.